oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -127,6 +128,28 @@ public ServiceAccountCredentials(
   public ServiceAccountCredentials(
       String clientId, String clientEmail, PrivateKey privateKey, String privateKeyId,
       Collection<String> scopes, HttpTransportFactory transportFactory, URI tokenServerUri) {
+    this(clientId, clientEmail, privateKey, privateKeyId, scopes, transportFactory, tokenServerUri, null);
+  }
+
+  /**
+   * Constructor with minimum identifying information and custom HTTP transport.
+   *
+   * @param clientId Client ID of the service account from the console. May be null.
+   * @param clientEmail Client email address of the service account from the console.
+   * @param privateKey RSA private key object for the service account.
+   * @param privateKeyId Private key identifier for the service account. May be null.
+   * @param scopes Scope strings for the APIs to be called. May be null or an empty collection,
+   *        which results in a credential that must have createScoped called before use.
+   * @param transportFactory HTTP transport factory, creates the transport used to get access
+   *        tokens.
+   * @param tokenServerUri URI of the end point that provides tokens.
+   * @param serviceAccountUser Email of the user account to impersonate, if delegating domain-wide
+   *        authority to the service account.
+   */
+  public ServiceAccountCredentials(
+      String clientId, String clientEmail, PrivateKey privateKey, String privateKeyId,
+      Collection<String> scopes, HttpTransportFactory transportFactory, URI tokenServerUri,
+      String serviceAccountUser) {
```

Comment:
That's exactly what I am aiming for. If there are 10 candidates for package-private vs public, and only 1 of them is requested by anyone, we have avoided making 9 more methods public, which is a big win in my mind.